### PR TITLE
chore: Bumping fastapi + starlette

### PR DESCRIPTION
**Pull request by bushwhackr**
**What this PR does / why we need it**:
Bumping vuln fastapi version. https://nvd.nist.gov/vuln/detail/CVE-2024-24762

**Changes**
+ cicd: mypy now only checks `sdk/python/feast` directory
+ cicd: sdk/python/requirements/*.txt are all bumped wrt. the version constraint in setup.py
+ lint: many base class with abstract methods have be altered to throw NotImplementedError instead due to missing implementation in some subclass.

**Which issue(s) this PR fixes**:

Fixes #3931 #3930

### CODE REVIEW

**Comment by sudohainguyen** [comment]
Linting failed, could you check @bushwhackr ?

**Comment by bushwhackr** [comment]
image

The issue is our mypy version I believe

**Comment by sudohainguyen** [comment]
Considering upgrading mypy too? 👀

**Comment by bushwhackr** [comment]
I'm looking to pin the mypy version to 1.4.1 but like the #3942 there are quite a number of linting failures. Let me see if I can do one with minimum changes.

**Comment by bushwhackr** [comment]
I'm of the opinion to skip integration tests mypy type errors till a later PR. Seems like alot of classes in sdk/python/tests/integration/feature_repos/universal/**/* have classes that do not confirm to the BaseClass abstract method definition.

**Comment by bushwhackr** [comment]
@sudohainguyen could you rerun the tests. Linting should pass now
```
   ~/repo/feast    bushwhackr/fastapi-starlette ?1 ❯ make lint-python                                                                                                                                                                                                                                                                      30s   feast   00:14:46
cd /home/chester/repo/feast/sdk/python; python -m mypy --exclude=/tests/ --follow-imports=skip feast
feast/on_demand_feature_view.py:345: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/on_demand_feature_view.py:368: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/on_demand_feature_view.py:403: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/feature_store.py:1009: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/repo_operations.py:249: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/repo_operations.py:263: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/utils/snowflake/snowflake_utils.py:78: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/materialization/contrib/bytewax/bytewax_materialization_dataflow.py:56: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Success: no issues found in 293 source files
cd /home/chester/repo/feast/sdk/python; python -m isort feast/ tests/ --check-only
Skipped 1 files
cd /home/chester/repo/feast/sdk/python; python -m flake8 feast/ tests/
cd /home/chester/repo/feast/sdk/python; python -m black --check feast tests
All done! ✨ 🍰 ✨
365 files would be left unchanged.
```

**Comment by sudohainguyen** [comment]
great! re-running

**Comment by sudohainguyen** [comment]
oh no 😢 @bushwhackr fail again

**Comment by bushwhackr** [comment]
hopefully it works now, just tested against python 3.8.

image

**Comment by sudohainguyen** [comment]
great, now unit tests 😓

**Comment by bushwhackr** [comment]
@sudohainguyen my bad, please try again

**Comment by sudohainguyen** [comment]
Perfect! All checks passed

**Review by sudohainguyen — APPROVED** [review]
great work! all problem solved 
lgtm
is there any issue in terms of performance we should know? @bushwhackr

**Comment by bushwhackr** [comment]
> great work! all problem solved lgtm is there any issue in terms of performance we should know? @bushwhackr

None at all, all changes are to fix linting issues.

File: Makefile [hunk 1]
```diff
@@ -310,7 +310,7 @@ format-python:
 	cd ${ROOT_DIR}/sdk/python; python -m black --target-version py38 feast tests
 
 lint-python:
-	cd ${ROOT_DIR}/sdk/python; python -m mypy
+	cd ${ROOT_DIR}/sdk/python; python -m mypy --exclude=/tests/ --follow-imports=skip feast
```
Comment (bushwhackr) [inline Makefile:313]: I would feel this is the biggest change, is that I only limit the linting checks to sdk/python/feast directory

Comment (sudohainguyen) [reply Makefile:313]: what if we still check tests? a lot of changes needed right?

**Comment by bushwhackr** [comment]
to revert to the old mypy settings:
```
feast/on_demand_feature_view.py:345: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/on_demand_feature_view.py:368: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/on_demand_feature_view.py:403: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/offline_stores/snowflake_source.py:222: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
feast/infra/offline_stores/snowflake_source.py:253: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
feast/feature_store.py:1009: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/repo_operations.py:249: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/repo_operations.py:263: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/utils/snowflake/snowflake_utils.py:78: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/offline_stores/snowflake.py:189: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
feast/infra/offline_stores/snowflake.py:240: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
feast/infra/offline_stores/snowflake.py:274: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
feast/infra/offline_stores/snowflake.py:395: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
tests/utils/feature_records.py:335: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/unit/test_registry_server.py:22: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/unit/test_registry_server.py:33: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/unit/test_registry_server.py:51: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/materialization/contrib/bytewax/bytewax_materialization_dataflow.py:56: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/integration/feature_repos/universal/online_store/redis.py:16: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/mysql.py:21: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/hbase.py:16: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/hazelcast.py:32: error: Return type "Dict[str, Any]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/dynamodb.py:18: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/datastore.py:26: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/cassandra.py:35: error: Return type "Dict[str, object]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/online_store/bigtable.py:31: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/feature_repos/universal/data_sources/snowflake.py:54: error: Argument 6 of "create_data_source" is incompatible with supertype "DataSourceCreator"; supertype defines the argument type as "Optional[str]"  [override]
tests/integration/feature_repos/universal/data_sources/snowflake.py:54: note: This violates the Liskov substitution principle
tests/integration/feature_repos/universal/data_sources/snowflake.py:54: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
tests/integration/feature_repos/universal/data_sources/snowflake.py:59: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOfflineStoreConfig"; expected "str"  [arg-type]
tests/integration/feature_repos/universal/data_sources/redshift.py:54: error: Argument 6 of "create_data_source" is incompatible with supertype "DataSourceCreator"; supertype defines the argument type as "Optional[str]"  [override]
tests/integration/feature_repos/universal/data_sources/redshift.py:54: note: This violates the Liskov substitution principle
tests/integration/feature_repos/universal/data_sources/redshift.py:54: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
tests/integration/feature_repos/universal/data_sources/file.py:38: error: Signature of "create_data_source" incompatible with supertype "DataSourceCreator"  [override]
tests/integration/feature_repos/universal/data_sources/file.py:38: note:      Superclass:
tests/integration/feature_repos/universal/data_sources/file.py:38: note:          def create_data_source(self, df: Any, destination_name: str, event_timestamp_column: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., timestamp_field: Optional[str] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/file.py:38: note:      Subclass:
tests/integration/feature_repos/universal/data_sources/file.py:38: note:          def create_data_source(self, df: Any, destination_name: str, timestamp_field: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/file.py:93: error: Signature of "create_data_source" incompatible with supertype "DataSourceCreator"  [override]
tests/integration/feature_repos/universal/data_sources/file.py:93: note:      Superclass:
tests/integration/feature_repos/universal/data_sources/file.py:93: note:          def create_data_source(self, df: Any, destination_name: str, event_timestamp_column: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., timestamp_field: Optional[str] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/file.py:93: note:      Subclass:
tests/integration/feature_repos/universal/data_sources/file.py:93: note:          def create_data_source(self, df: Any, destination_name: str, timestamp_field: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/file.py:167: error: Signature of "create_data_source" incompatible with supertype "DataSourceCreator"  [override]
tests/integration/feature_repos/universal/data_sources/file.py:167: note:      Superclass:
tests/integration/feature_repos/universal/data_sources/file.py:167: note:          def create_data_source(self, df: Any, destination_name: str, event_timestamp_column: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., timestamp_field: Optional[str] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/file.py:167: note:      Subclass:
tests/integration/feature_repos/universal/data_sources/file.py:167: note:          def create_data_source(self, df: Any, destination_name: Optional[str] = ..., suffix: Optional[str] = ..., timestamp_field: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/bigquery.py:63: error: Signature of "create_data_source" incompatible with supertype "DataSourceCreator"  [override]
tests/integration/feature_repos/universal/data_sources/bigquery.py:63: note:      Superclass:
tests/integration/feature_repos/universal/data_sources/bigquery.py:63: note:          def create_data_source(self, df: Any, destination_name: str, event_timestamp_column: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., timestamp_field: Optional[str] = ...) -> DataSource
tests/integration/feature_repos/universal/data_sources/bigquery.py:63: note:      Subclass:
tests/integration/feature_repos/universal/data_sources/bigquery.py:63: note:          def create_data_source(self, df: Any, destination_name: str, timestamp_field: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., **kwargs: Any) -> DataSource
feast/infra/online_stores/snowflake.py:117: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOnlineStoreConfig"; expected "str"  [arg-type]
feast/infra/online_stores/snowflake.py:181: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOnlineStoreConfig"; expected "str"  [arg-type]
feast/infra/online_stores/snowflake.py:223: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOnlineStoreConfig"; expected "str"  [arg-type]
feast/infra/online_stores/snowflake.py:251: error: Argument 1 to "GetSnowflakeConnection" has incompatible type "SnowflakeOnlineStoreConfig"; expected "str"  [arg-type]
feast/infra/offline_stores/contrib/spark_offline_store/tests/data_source.py:67: error: Signature of "create_data_source" incompatible with supertype "DataSourceCreator"  [override]
feast/infra/offline_stores/contrib/spark_offline_store/tests/data_source.py:67: note:      Superclass:
feast/infra/offline_stores/contrib/spark_offline_store/tests/data_source.py:67: note:          def create_data_source(self, df: Any, destination_name: str, event_timestamp_column: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., timestamp_field: Optional[str] = ...) -> DataSource
feast/infra/offline_stores/contrib/spark_offline_store/tests/data_source.py:67: note:      Subclass:
feast/infra/offline_stores/contrib/spark_offline_store/tests/data_source.py:67: note:          def create_data_source(self, df: Any, destination_name: str, timestamp_field: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., **kwargs: Any) -> DataSource
feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py:58: error: Return type "MsSqlServerOfflineStoreConfig" of "create_offline_store_config" incompatible with return type "FeastConfigBaseModel" in supertype "DataSourceCreator"  [override]
feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py:63: error: Signature of "create_data_source" incompatible with supertype "DataSourceCreator"  [override]
feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py:63: note:      Superclass:
feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py:63: note:          def create_data_source(self, df: Any, destination_name: str, event_timestamp_column: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., timestamp_field: Optional[str] = ...) -> DataSource
feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py:63: note:      Subclass:
feast/infra/offline_stores/contrib/mssql_offline_store/tests/data_source.py:63: note:          def create_data_source(self, df: Any, destination_name: str, timestamp_field: Any = ..., created_timestamp_column: Any = ..., field_mapping: Optional[Dict[str, str]] = ..., **kwargs: Any) -> DataSource
feast/infra/offline_stores/contrib/athena_offline_store/tests/data_source.py:54: error: Argument 6 of "create_data_source" is incompatible with supertype "DataSourceCreator"; supertype defines the argument type as "Optional[str]"  [override]
feast/infra/offline_stores/contrib/athena_offline_store/tests/data_source.py:54: note: This violates the Liskov substitution principle
feast/infra/offline_stores/contrib/athena_offline_store/tests/data_source.py:54: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
feast/infra/contrib/spark_kafka_processor.py:71: error: Return type "StreamingQuery" of "ingest_stream_feature_view" incompatible with return type "None" in supertype "StreamProcessor"  [override]
feast/infra/contrib/spark_kafka_processor.py:132: error: Return type "StreamingQuery" of "_write_stream_data" incompatible with return type "None" in supertype "StreamProcessor"  [override]
tests/integration/registration/test_universal_types.py:337: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/integration/registration/test_universal_types.py:356: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/integration/feature_repos/repo_configuration.py:103: error: Dict entry 0 has incompatible type "str": "Tuple[str, Type[FileDataSourceCreator]]"; expected "str": "DataSourceCreator"  [dict-item]
tests/integration/feature_repos/repo_configuration.py:104: error: Dict entry 1 has incompatible type "str": "Tuple[str, Type[BigQueryDataSourceCreator]]"; expected "str": "DataSourceCreator"  [dict-item]
tests/integration/feature_repos/repo_configuration.py:105: error: Dict entry 2 has incompatible type "str": "Tuple[str, Type[RedshiftDataSourceCreator]]"; expected "str": "DataSourceCreator"  [dict-item]
tests/integration/feature_repos/repo_configuration.py:106: error: Dict entry 3 has incompatible type "str": "Tuple[str, Type[SnowflakeDataSourceCreator]]"; expected "str": "DataSourceCreator"  [dict-item]
tests/integration/feature_repos/repo_configuration.py:172: error: Value expression in dictionary comprehension has incompatible type "Tuple[Union[str, Dict[Any, Any], None], Optional[Type[OnlineStoreCreator]]]"; expected type "Tuple[Union[str, Dict[str, str]], Optional[Type[OnlineStoreCreator]]]"  [misc]
tests/integration/feature_repos/repo_configuration.py:353: error: Argument "driver_odfv" to "UniversalFeatureViews" has incompatible type "Optional[OnDemandFeatureView]"; expected "OnDemandFeatureView"  [arg-type]
tests/integration/feature_repos/repo_configuration.py:354: error: Argument 1 to "conv_rate_plus_100_feature_view" has incompatible type "List[Any]"; expected "Dict[str, Union[RequestSource, FeatureView]]"  [arg-type]
tests/integration/feature_repos/repo_configuration.py:378: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/integration/feature_repos/repo_configuration.py:415: error: Incompatible types in assignment (expression has type "FeastConfigBaseModel", variable has type "Union[str, Dict[Any, Any], None]")  [assignment]
tests/integration/feature_repos/repo_configuration.py:418: error: Incompatible types in assignment (expression has type "Union[str, Dict[Any, Any], None]", variable has type "FeastConfigBaseModel")  [assignment]
tests/integration/feature_repos/repo_configuration.py:425: error: Missing named argument "feature_logging" for "AwsLambdaFeatureServerConfig"  [call-arg]
tests/integration/feature_repos/repo_configuration.py:434: error: Incompatible types in assignment (expression has type "LocalFeatureServerConfig", variable has type "AwsLambdaFeatureServerConfig")  [assignment]
tests/integration/feature_repos/repo_configuration.py:449: error: Missing named argument "registry_store_type" for "RegistryConfig"  [call-arg]
tests/integration/feature_repos/repo_configuration.py:449: error: Missing named argument "s3_additional_kwargs" for "RegistryConfig"  [call-arg]
tests/conftest.py:222: error: Incompatible types in assignment (expression has type "List[Tuple[str, Type[DataSourceCreator]]]", variable has type "Optional[List[Optional[DataSourceCreator]]]")  [assignment]
tests/conftest.py:225: error: List item 0 has incompatible type "Tuple[str, Type[FileDataSourceCreator]]"; expected "Optional[DataSourceCreator]"  [list-item]
tests/conftest.py:237: error: Incompatible types in assignment (expression has type "dict_values[str, Tuple[Union[str, Dict[str, str]], Optional[Type[OnlineStoreCreator]]]]", variable has type "Optional[List[Optional[Tuple[Union[str, Dict[str, str]], Optional[Type[OnlineStoreCreator]]]]]]")  [assignment]
tests/conftest.py:262: error: "DataSourceCreator" object is not iterable  [misc]
tests/conftest.py:262: error: "None" object is not iterable  [misc]
tests/conftest.py:263: error: "None" object is not iterable  [misc]
tests/conftest.py:292: error: List item 0 has incompatible type "Tuple[str, Type[FileDataSourceCreator]]"; expected "Optional[DataSourceCreator]"  [list-item]
tests/unit/online_store/test_online_retrieval.py:139: error: Missing named argument "registry_store_type" for "RegistryConfig"  [call-arg]
tests/unit/online_store/test_online_retrieval.py:139: error: Missing named argument "s3_additional_kwargs" for "RegistryConfig"  [call-arg]
tests/unit/online_store/test_online_retrieval.py:202: error: Missing named argument "registry_store_type" for "RegistryConfig"  [call-arg]
tests/unit/online_store/test_online_retrieval.py:202: error: Missing named argument "s3_additional_kwargs" for "RegistryConfig"  [call-arg]
tests/unit/infra/test_local_registry.py:41: error: Missing named argument "registry_store_type" for "RegistryConfig"  [call-arg]
tests/unit/infra/test_local_registry.py:41: error: Missing named argument "s3_additional_kwargs" for "RegistryConfig"  [call-arg]
tests/unit/cli/test_cli_chdir.py:21: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/unit/cli/test_cli_chdir.py:24: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/unit/cli/test_cli_chdir.py:28: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/unit/cli/test_cli_chdir.py:37: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/unit/cli/test_cli_chdir.py:47: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/unit/cli/test_cli_chdir.py:52: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/unit/cli/test_cli_chdir.py:55: error: List item 1 has incompatible type "Path"; expected "str"  [list-item]
tests/integration/registration/test_registry.py:45: error: Missing named argument "registry_store_type" for "RegistryConfig"  [call-arg]
tests/integration/registration/test_registry.py:45: error: Missing named argument "s3_additional_kwargs" for "RegistryConfig"  [call-arg]
tests/integration/registration/test_registry.py:56: error: Missing named argument "registry_store_type" for "RegistryConfig"  [call-arg]
tests/integration/registration/test_registry.py:56: error: Missing named argument "s3_additional_kwargs" for "RegistryConfig"  [call-arg]
tests/integration/offline_store/test_universal_historical_retrieval.py:170: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/integration/offline_store/test_s3_custom_endpoint.py:43: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
tests/integration/offline_store/test_offline_write.py:134: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
feast/infra/offline_stores/contrib/trino_offline_store/tests/data_source.py:87: error: Argument 6 of "create_data_source" is incompatible with supertype "DataSourceCreator"; supertype defines the argument type as "Optional[str]"  [override]
feast/infra/offline_stores/contrib/trino_offline_store/tests/data_source.py:87: note: This violates the Liskov substitution principle
feast/infra/offline_stores/contrib/trino_offline_store/tests/data_source.py:87: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
feast/infra/offline_stores/contrib/trino_offline_store/tests/data_source.py:125: error: Missing named argument "user" for "TrinoOfflineStoreConfig"  [call-arg]
feast/infra/offline_stores/contrib/trino_offline_store/tests/data_source.py:125: error: Missing named argument "auth" for "TrinoOfflineStoreConfig"  [call-arg]
feast/infra/offline_stores/contrib/postgres_offline_store/tests/data_source.py:88: error: Argument 6 of "create_data_source" is incompatible with supertype "DataSourceCreator"; supertype defines the argument type as "Optional[str]"  [override]
feast/infra/offline_stores/contrib/postgres_offline_store/tests/data_source.py:88: note: This violates the Liskov substitution principle
feast/infra/offline_stores/contrib/postgres_offline_store/tests/data_source.py:88: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
feast/infra/offline_stores/contrib/postgres_offline_store/tests/data_source.py:109: error: Return type "Dict[str, str]" of "create_online_store" incompatible with return type "FeastConfigBaseModel" in supertype "OnlineStoreCreator"  [override]
tests/integration/e2e/test_python_feature_server.py:131: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 73 errors in 27 files (checked 427 source files)
```

**Comment by sudohainguyen** [comment]
alright, let's get rid of it in another PR

File: sdk/python/feast/infra/registry/snowflake.py [hunk 2]
```diff
@@ -418,7 +418,7 @@ def _delete_object(
             """
             cursor = execute_snowflake_statement(conn, query)
 
-            if cursor.rowcount < 1 and not_found_exception:
+            if cursor.rowcount < 1 and not_found_exception:  # type: ignore
```
Comment (shuchu) [inline sdk/python/feast/infra/registry/snowflake.py:421]: I tried this and can pass the mypy check:
`  if cursor.rowcount and (cursor.rowcount < 1) and not_found_exception:`

Comment (bushwhackr) [reply sdk/python/feast/infra/registry/snowflake.py:421]: You're right, this is due to another change I did to the mypy config: `--follow-imports=skip`. 

run the below command to see the error:
python -m mypy --exclude=/tests/ sdk/python/feast/infra/registry/snowflake.py you will see an error.
```
`sdk/python/feast/infra/registry/snowflake.py:421: error: Unsupported operand types for > ("int" and "None")  [operator]
```

Comment (bushwhackr) [reply sdk/python/feast/infra/registry/snowflake.py:421]: essentially cursor.rowcount could return none from snowflake's signature
```
    @property
    def rowcount(self) -> int | None:
        return self._total_rowcount if self._total_rowcount >= 0 else None
 ```

**Comment by bushwhackr** [comment]
any blockers?

**Comment by sudohainguyen** [comment]
Need your merge actions here 🙏 @franciscojavierarceo

**Comment by shuchu** [comment]
> BaseClass

I will fix (have fixed) this in my PR. https://github.com/feast-dev/feast/pull/3942

**Pull request merged**